File: angular-portfolio/src/app/models/Imagen.ts
class Imagen {
    protected _ruta: string = '';
    protected _archivo: string = '';

    public constructor() {}

    public get ruta(): string {
        return this._ruta;
    }
    public set ruta(value: string) {
        this._ruta = value;
    }

    public get archivo(): string {
        return this._archivo;
    }
    public set archivo(value: string) {
        this._archivo = value;
    }
}

File: api/src/main/java/com/facundocastro/api/model/Imagen.java
package com.facundocastro.api.model;

import com.sun.istack.NotNull;
import lombok.Data;

import javax.persistence.Embeddable;

@Data
@Embeddable
public class Imagen {
    @NotNull
    private String ruta;
    @NotNull
    private String archivo;
}
